pkg/controller/nodes/executor.go

```diff
@@ -796,12 +796,7 @@ func isTimeoutExpired(queuedAt *metav1.Time, timeout time.Duration) bool {
 
 func (c *nodeExecutor) isEligibleForRetry(nCtx interfaces.NodeExecutionContext, nodeStatus v1alpha1.ExecutableNodeStatus, err *core.ExecutionError) (currentAttempt, maxAttempts uint32, isEligible bool) {
 	if config.GetConfig().NodeConfig.IgnoreRetryCause {
-		currentAttempt = nodeStatus.GetAttempts() + 1 + nodeStatus.GetSystemFailures()
-		maxAttempts = uint32(config.GetConfig().NodeConfig.DefaultMaxAttempts)
-		if nCtx.Node().GetRetryStrategy() != nil && nCtx.Node().GetRetryStrategy().MinAttempts != nil && *nCtx.Node().GetRetryStrategy().MinAttempts != 0 {
-			maxAttempts = uint32(*nCtx.Node().GetRetryStrategy().MinAttempts)
-		}
-		isEligible = currentAttempt < maxAttempts
+		currentAttempt = nodeStatus.GetAttempts() + 1
```

Comment:
We removed `nodeStatus.GetSystemFailures()`. As you can see in
```
currentAttempt = (nodeStatus.GetAttempts() + 1) - nodeStatus.GetSystemFailures()
```
(line 813)

`GetAttempts()` is the sum of system failures and user failures

When creating the [test table](https://github.com/flyteorg/flyte/pull/3902#issuecomment-1690174655) we did not notice this as we always only used a single retry for system failures. For our final tests we now did some more extensive search for one off errors.